src/testing/tree_construction.rs

```diff
@@ -137,20 +137,25 @@ impl Test {
     }
 
     /// Runs the test and returns the result
-    pub fn run(&self) -> Result<TestResult> {
-        let (actual_document, actual_errors) = self.parse()?;
-        let root = self.match_document_tree(&actual_document.get());
-
-        Ok(TestResult {
-            root,
-            actual_document,
-            actual_errors,
-        })
+    pub fn run(&self) -> Result<Vec<TestResult>> {
+        let mut results = vec![];
+
+        for &scripting_enabled in self.script_modes() {
+            let (actual_document, actual_errors) = self.parse(scripting_enabled)?;
+            let root = self.match_document_tree(&actual_document.get());
+            results.push(TestResult {
+                root,
+                actual_document,
+                actual_errors,
+            });
+        }
```

Comment:
This is the heart of this PR.

Reply:
Didnt i implement this already? Or did i miss a case somewhere?

Reply:
[This is what was there up until recently](https://github.com/gosub-browser/gosub-engine/blob/581dd3501c7b760a035eaac306f34c75356f40fa/src/testing/tree_construction.rs#L414-L419).  It implies a single run for a test — either scripting is enabled, or it's not enabled.  By contrast, the spec implies 1-2 runs: [on, off, or both](https://github.com/html5lib/html5lib-tests/tree/master/tree-construction#tree-construction-tests).

This PR picks up the "both" case (2 iterations of the test), which we weren't doing before.